community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexAccessor.java

```diff
@@ -100,7 +102,10 @@ public IndexReader newReader()
     @Override
     public BoundedIterable<Long> newAllEntriesReader()
     {
-        BoundedIterable<Long>[] entries = instancesAs( BoundedIterable.class, IndexAccessor::newAllEntriesReader );
+        @SuppressWarnings( "unchecked" )
+        BoundedIterable<Long>[] entries = Arrays.stream( instancesAs( BoundedIterable.class, IndexAccessor::newAllEntriesReader ) )
+                .filter( notNull() )
```

Comment:
i think filtering out and guarding against nulls everywhere is a source of errors, what if instead we will have those instance as an empty instance of some sort?